sessions_core/01_introduction.qmd

```diff
@@ -0,0 +1,751 @@
+```
+
+```{r}
+#| echo: false
+#| eval: true
+# INTERNALS
+region <- c("Mandoul", "Moyen-Chari", "Logone Oriental",
+                      "Tibesti", "Logone Occidental")
+```
+
+
+We can now use functions on the objects we have created:
+
+```{r}
+#| eval: true
+mean(cases)      # calculate the mean value of the cases vector
+toupper(region)  # convert all the values in region to lower case
+```
+
+
+::: {.write}
+Use the function `sum()` to calculate the sum of `cases`
+:::
+
+```{r}
+#| include = solutions
+# SOLUTION
+sum(cases)
+```
+
+::: {.write}
+Pass `region` to the function `tolower()`
+:::
+
+```{r}
+#| include = solutions
+# SOLUTION
+tolower(region)
+```
+
+### Accessing the Values of a Vector
+
+It is possible to access the value of a vector using square brackets containing the index (position) of the desired value, ie: `[3]` or `[189]`.
+
+```{r}
+#| eval: true
+cases[2]   # 2nd value of cases
+cases[10]  # 10th value of cases
+```
+
+Ooops it does not exist! We will come back to what this `NA` means in @sec-missing-values.
+
+We can also access a *range* of values, just as we might do in Excel. To create a range we use the `:` operator to separate the desired minimum and maximum index:
+
+```{r}
+#| eval: true
+cases[2:4]  # 2nd to 4th values of cases
+```
+
+::: {.write}
+Get the 3rd value of `region`. Write code to access the values "Mandoul" and "Moyen-Chari" in the vector `region`.
+:::
+
+```{r}
+#| include = solutions
+# SOLUTION
+region[3]
+region[1:2]
+```
+
+
+### Dataframes
+
+*Dataframes* are a special type of list that is composed of multiple simple vectors of the same length. Dataframes take the shape of a table where each of these vectors is a named column, similar to a table in Excel. As epidemiologists, this type of data structure is perhaps the most useful and you will likely use them on a daily basis.
+
+#### Creating a Dataframe
+
+We can create a dataframe using the function `data.frame()`:
+
+```{r}
+#| eval: true
+data.frame(col1 = c(1, 4, 2, 9),
+           col2 = c("a bit of text", "some more text", "hello", "epidemiologists"))
+```
+
+::: {.write}
+In your script, create a dataframe called `data_cases` that contains `cases` in one column and `region` in the other.
+:::
+
+```{r}
+#| include = solutions
+# SOLUTION
+data_cases <- data.frame(
+  cases = c(2, 5, 8, 0, 4),
+  region = c("Mandoul", "Sud Kivu", "Kasai oriental", 
+               "Kasai", "Haut Katanga"))
+
+# OR
+data_cases <- data.frame(cases, region)
+```
+
+
+```{r}
+#| echo: false
+#| eval: true
+# INTERNAL
+data_cases <- data.frame(
+  cases = c(2, 5, 8, 0, 4),
+  region = c("Mandoul", "Sud Kivu", "Kasai oriental", 
+               "Kasai", "Haut Katanga"))
+```
+
+
+#### Exploring a Dataframe
+
+`data_cases` should now appear in your environment. You can click on its name to see some additional information, or click on the blue circle with a white triangle in it to open the object in the *Viewer*. 
```

Comment:
I don't think this is right? Clicking on the blue circle gives a summary of the data.frame (similar output to `str()`), and clicking on the name will open the data.frame in a new tab in the scripts panel. As far as I know, you don't look at the data in the Viewer tab.